cmd/up.go

```diff
@@ -142,3 +165,49 @@ func isGKE(disco discovery.DiscoveryInterface) (bool, error) {
 
 	return false, nil
 }
+
+func mongoSecretExists(c kubecfg.ApplyCmd, name, ns string) (*unstructured.Unstructured, bool, error) {
+	gvk := schema.GroupVersionKind{Version: "v1", Kind: "Secret"}
+	rc, err := clientForGroupVersionKind(c.ClientPool, c.Discovery, gvk, ns)
+	if err != nil {
+		return nil, false, err
+	}
+	pwFields := []string{"mongodb-password", "mongodb-root-password"}
+	prevSec, err := rc.Get(name)
+
+	if k8sErrors.IsNotFound(err) {
+		return nil, false, nil
+	}
+
+	if err == nil {
+		if prevSec.Object["data"] == nil {
+			return nil, true, fmt.Errorf("secret %s already exists but it doesn't contain any expected key", name)
+		}
+
+		prevPw := prevSec.Object["data"].(map[string]interface{})
+		for _, p := range pwFields {
+			if prevPw[p] == nil {
+				return nil, true, fmt.Errorf("secret %s already exists but it doesn't contain the expected key %s", name, p)
+			}
+		}
+
+		return prevSec, true, nil
+	}
+
+	return nil, true, err
```

Comment:
Again, invert return-early if/else clauses to keep a linear flow for the reader:

```golang
if k8sErrors.IsNotFound(err) {
    return nil, false, nil
}
if err != nil {   // <- I inverted this one and brought the early-return up here
    return nil, true, err
}

if prevSec.Object["data"] == nil {
    return nil, true, fmt.Errorf("secret already exists ...")
}

prevPw := prevSec.Object["data"].(map[string]interface{})
for _, p := range pwFields {
    ...
}

return prevSec, true, nil
```